src/SourceBuild/content/test/Microsoft.DotNet.UnifiedBuild.Tests/Microsoft.DotNet.UnifiedBuild.Tests.csproj

```diff
@@ -0,0 +1,52 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>$(NetCurrent)</TargetFramework>
+    <Nullable>enable</Nullable>
+    <DefaultExcludesInProjectFolder>$(DefaultExcludesInProjectFolder);assets/**/*</DefaultExcludesInProjectFolder>
+    <UBTestsWarnOnSdkContentDiffs>true</UBTestsWarnOnSdkContentDiffs>
+    <VSTestLogger>console%3bverbosity=diagnostic;trx%3bverbosity=diagnostic%3bLogFileName=Microsoft.DotNet.UnifiedBuild.Tests.xml</VSTestLogger>
+    <VSTestResultsDirectory>$(ArtifactsTestResultsDir)</VSTestResultsDirectory>
+    <VsTestUseMSBuildOutput>true</VsTestUseMSBuildOutput>
+    <!-- DOTNET_ROOT is not set when running the test, so the testhost looks in the global installation location for preview framework assemblies, which aren't there -->
+    <!-- As a workaround, just publish self-contained to copy the framework assemblies from the local sdk -->
+    <SelfContained>true</SelfContained>
+    <RuntimeIdentifier>$([System.Runtime.InteropServices.RuntimeInformation]::RuntimeIdentifier)</RuntimeIdentifier>
```

Comment:
Sounds like this test project should always use the repo local SDK folder. I would set the `DotNetHostPath` as a .runsettings setting instead to achieve that: https://github.com/dotnet/runtime/blob/59d9749a225d27ef27aaa039ff561f3c3c7a6b83/eng/testing/.runsettings#L20